internal/services/applications/application_resource.go

```diff
@@ -266,6 +266,58 @@ func applicationResource() *pluginsdk.Resource {
 				},
 			},
 
+			"password": {
+				Description: "App password definition",
+				Type:        pluginsdk.TypeSet,
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"display_name": {
+							Description: "A display name for the password",
+							Type:        pluginsdk.TypeString,
+							Required:    true,
+						},
+
+						"start_date": {
+							Description:  "The start date from which the password is valid, formatted as an RFC3339 date string (e.g. `2018-01-01T01:02:03Z`). If this isn't specified, the current date is used",
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+
+						"end_date": {
+							Description:  "The end date until which the password is valid, formatted as an RFC3339 date string (e.g. `2018-01-01T01:02:03Z`)",
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.IsRFC3339Time,
+						},
+
+						"end_date_relative": {
+							Description:  "A relative duration for which the password is valid until, for example `240h` (10 days) or `2400h30m`. Changing this field forces a new resource to be created",
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
```

Comment:
I think we should remove `end_date_relative`, since users can use the `timeadd()` Terraform function to compute this, and it is problematic when persisting state in cases where the `end_date` value is changed and a password has to be replaced.

```suggestion
```